tests/foreman/longrun/test_oscap.py

```diff
@@ -42,670 +42,593 @@
-        scap_id = default_content['id']
-        scap_profile_ids = [
-            profile['id']
-            for profile in default_content['scap-content-profiles']
-            if scap_profile in profile['title']
-        ]
-        return scap_id, scap_profile_ids
-
-    @classmethod
-    def configure_puppet_test(cls):
-        """Sets up the whole provisioning environment needed for Puppet based
-        end-to-end tests like OSCAP etc
-
-        :returns: A dict of entities to help with provisioning
-        """
-        cls.rhel6_content = OSCAP_DEFAULT_CONTENT['rhel6_content']
-        cls.rhel7_content = OSCAP_DEFAULT_CONTENT['rhel7_content']
-        cls.rhel8_content = OSCAP_DEFAULT_CONTENT['rhel8_content']
-        sat6_hostname = settings.server.hostname
-        proxy = Proxy.list({'search': sat6_hostname})[0]
-        p_features = set(proxy.get('features').split(', '))
-        if {'Puppet', 'Ansible', 'Openscap'}.issubset(p_features):
-            cls.proxy_id = proxy.get('id')
-        else:
-            raise ProxyError('Some features like Puppet, DHCP, Openscap, Ansible are not present')
-        ak_name_8 = gen_string('alpha')
-        ak_name_7 = gen_string('alpha')
-        ak_name_6 = gen_string('alpha')
-        repo_values = [
-            {'repo': settings.sattools_repo['rhel8'], 'akname': ak_name_8},
-            {'repo': settings.sattools_repo['rhel7'], 'akname': ak_name_7},
-            {'repo': settings.sattools_repo['rhel6'], 'akname': ak_name_6},
-        ]
-        # Create new organization and environment.
-        org = entities.Organization(name=gen_string('alpha')).create()
-        cls.puppet_env = (
-            entities.Environment().search(query={'search': 'name=production'})[0].read()
-        )
-        cls.puppet_env.organization.append(org)
-        cls.puppet_env = cls.puppet_env.update(['organization'])
-        smart_proxy = (
-            entities.SmartProxy().search(query={'search': f'name={sat6_hostname}'})[0].read()
+def fetch_scap_and_profile_id(scap_name, scap_profile):
+    """Extracts the scap ID and scap profile id
+
+    :param scap_name: Scap title
+    :param scap_profile: Scap profile you want to select
+
+    :returns: scap_id and scap_profile_id
+    """
+
+    default_content = Scapcontent.info({'title': scap_name}, output_format='json')
+    scap_id = default_content['id']
+    scap_profile_ids = [
+        profile['id']
+        for profile in default_content['scap-content-profiles']
+        if scap_profile in profile['title']
+    ]
+    return scap_id, scap_profile_ids
+
+
+@pytest.fixture(scope='module')
+def scap_setup(request, module_org):
+    """Sets up the whole provisioning environment needed for Puppet based
+    end-to-end tests like OSCAP etc
+
+    :returns: A dict of entities to help with provisioning
+    """
+    request.puppet_classes = ['foreman_scap_client::params', 'foreman_scap_client']
```

Comment:
I don't think this should be tied to the request object, its static. It can be a module level constant.

Reply:
updated pr